cli/azd/pkg/ai/python_bridge.go

```diff
@@ -0,0 +1,130 @@
+package ai
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/python"
+	"github.com/azure/azure-dev/cli/azd/resources"
+)
+
+// ScriptPath is a type to represent the path of a Python script
+type ScriptPath string
+
+const (
+	// PromptFlowClient is the path to the PromptFlow Client Python script
+	PromptFlowClient ScriptPath = "pf_client.py"
+	// MLClient is the path to the ML Client Python script
+	MLClient ScriptPath = "ml_client.py"
+)
+
+// PythonBridge is an interface to execute python components from the embedded AI resources project
+type PythonBridge interface {
+	Initialize(ctx context.Context) error
+	RequiredExternalTools(ctx context.Context) []tools.ExternalTool
+	Run(ctx context.Context, scriptName ScriptPath, args ...string) (*exec.RunResult, error)
+}
```

Comment:
can we avoid interface here?

Reply:
I don't see why we would avoid interfaces. Interfaces make it easier to create mock implementations for testing purposes, facilitating unit testing in Go.

Reply:
design can't be driven by what makes it easier to test the system.

Let's use interfaces on top of multiple-implementations, as a contract like it is meant to be.

Design decisions should benefit the system itself; not the way the system is tested.